pootle/apps/virtualfolder/utils.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import os
+
+from pootle.core.url_helpers import split_pootle_path
+from pootle_store.models import Store
+
+
+def split_vfolder_location(path):
+    (lang_code, proj_code,
+     dir_path, filename) = split_pootle_path(path)
+    if lang_code == "{LANG}":
+        lang_code = None
+    if proj_code == "{PROJ}":
+        proj_code = None
+    return lang_code, proj_code, dir_path, filename
+
+
+class VirtualFolderStoreMatcher(object):
+
+    def __init__(self, vf):
+        self.vf = vf
+
+    def add_related_stores(self):
```

Comment:
just thinking we might need to change this method to `update_related_stores` and include removing no longer related stores